doc/articles/Surfaces.md

```diff
@@ -0,0 +1,250 @@
+    return;
+}
+
+// Do things with capabilities
+bool canSampleSurface = caps.usages & WGPUTextureUsage_TextureBinding;
+WGPUTextureFormat preferredFormat = caps.format[0];
+
+bool supportsMailbox = false;
+for (size_t i = 0; i < caps.presentModeCount; i++) {
+    if (caps.presentModes[i] == WGPUPresentMode_Mailbox) supportsMailbox = true;
+}
+
+// Cleanup
+wgpuSurfaceCapabilitiesFreeMembers(&caps);
+```
+
+The behavior of `::wgpuSurfaceGetCapabilities``(surface, adapter, caps)` is:
+
+ - If any of these validation steps fails, return false. (TODO return an error WGPUStatus):
+
+   - Validate that all the sub-descriptors in the chain for `caps` are known to this implementation.
+   - Validate that `surface` and `adapter` are created from the same @ref WGPUInstance.
+
+ - Fill `caps` with `adapter`'s capabilities to render to `surface`.
+ - Return true. (TODO return WGPUStatus_Success)
+
+## Surface Configuration {#Surface-Configuration}
+
+Before it can use it for rendering, the application must configure the surface.
+The configuration is the second step of the negotiation, done after analyzing the results of `::wgpuSurfaceGetCapabilities`.
+It contains the following kinds of parameters:
+
+ - The @ref WGPUDevice that will be used to render to the surface.
+ - Parameters for the textures returned by `::wgpuSurfaceGetCurrentTexture`.
+ - @ref WGPUPresentMode and @ref WGPUCompositeAlphaMode parameters for how and when the surface will be presented to the user.
+
+This is an example of how to configure a @ref WGPUSurface:
+
+```c
+WGPUSurfaceConfiguration config = {
+    nextInChain = nullptr,
+    device = myDevice,
+    format = WGPUTextureFormat_BGRA8Unorm,
+    width = 640, // Depending on the window size.
+    height = 480,
+    usage = WGPUTextureUsage_RenderAttachment,
+    presentMode = supportsMailbox ? WGPUPresentMode_Mailbox : WGPUPresentMode_Fifo,
+    alphaMode = WGPUCompositeAlphaMode_Auto,
+};
+wgpuSurfaceConfigure(mySurface, &config);
+```
+
+The parameters for the texture are used to create the texture each frame (see @ref Surface-Presenting) with the equivalent @ref WGPUTextureDescriptor computed like this:
+
+```c
+WGPUTextureDescriptor GetSurfaceEquivalentTextureDescriptor(const WGPUSurfaceConfiguration* config) {
+    return {
+        // Parameters controlled by the surface configuration.
+        .size = {config->width, config->height, 1},
+        .usage = config->usage,
+        .format = config->format,
+        .viewFormatCount = config->viewFormatCount,
+        .viewFormats = config->viewFormats,
+
+        // Parameters that cannot be changed.
+        .nextInChain = nullptr,
+        .label = "",
+        .dimension = WGPUTextureDimension_2D,
+        .sampleCount = 1,
+        .mipLevelCount = 1,
+    };
+}
+```
+
+When successfully a surface is successfully configured, the new configuration overrides any previous configuration and destroys the previous current texture (if any) so it can no longer be used.
```

Comment:
```suggestion
When a surface is successfully configured, the new configuration overrides any previous configuration and destroys the previous current texture (if any) so it can no longer be used.
```